spyder/utils/introspection/utils.py

```diff
@@ -210,6 +210,45 @@ def get_keywords(lexer):
                 continue
     return keywords
 
+def get_words(file_path=None, content=None, extension=None):
+    """
+    Extract all words from a source code file to be used in code completion.
+
+    Extract the list of words that contains the file in the editor,
+    to carry out the inline completion similar to VSCode.
+    """
+    if (file_path is None and (content is None or extension is None) or
+                    file_path and content and extension):
+        error_msg = ('Must provide `file_path` or `content` and `extension`')
+        raise Exception(error_msg)
+
+    if file_path and content is None and extension is None:
+
+        ext = os.path.splitext(file_path)[1]
+        if ext in ['.css']:
+            regex = re.compile(r'([^a-zA-Z-])')
+        elif ext in ['.R', '.c', 'md', '.cpp, java','.py']:
+            regex = re.compile(r'([^a-zA-Z_])')
+        else:
+            regex = re.compile(r'([^a-zA-Z])')
+
+        with open(file_path, 'r') as infile:
+            lines = [regex.sub(r' ', line).split() for line in infile]
+        words = list(set([x for words in lines for x in words if x != []]))
+        return words
+
+    if extension in ['.css']:
+        regex = re.compile(r'([^a-zA-Z-])')
+    elif extension in ['.R', '.c', 'md', '.cpp, java', '.py']:
+        regex = re.compile(r'([^a-zA-Z_])')
+    else:
+        regex = re.compile(r'([^a-zA-Z])')
+
+    # lines = [regex.sub(r' ', content).split()]
+    # words = list(set([x for words in lines for x in words if x != []]))
+    words = sorted(set(regex.sub(r' ', content).split()))
+
+    return words
```

Comment:
There is duplicate code and we dont need to do this. 

This code is in two different parts of the method.

```python
+    if extension in ['.css']:
 +        regex = re.compile(r'([^a-zA-Z-])')
 +    elif extension in ['.R', '.c', 'md', '.cpp, java', '.py']:
 +        regex = re.compile(r'([^a-zA-Z_])')
 +    else:
 +        regex = re.compile(r'([^a-zA-Z])')
```

What I suggest to fix this is

```python

def get_words_file(file_path=None, content=None, extension=None):
    """
    Extract all words from source code string.

    Extract the list of words that contains the file in the editor,
    to carry out the inline completion similar to VSCode.
    """
    if (file_path is None and (content is None or extension is None) or
                    file_path and content and extension):
        error_msg = ('Must provide `file_path` or `content` and `extension`')
        raise Exception(error_msg)

    if file_path and content is None and extension is None:
        extension = os.path.splitext(file_path)[1]

        with open(file_path, 'r') as infile:
            content = infile.read()

    if extension in ['.css']:
        regex = re.compile(r'([^a-zA-Z-])')
    elif extension in ['.R', '.c', 'md', '.cpp, java', '.py']:
        regex = re.compile(r'([^a-zA-Z_])')
    else:
        regex = re.compile(r'([^a-zA-Z])')

    words = sorted(set(regex.sub(r' ', content).split()))
    return words
```